scapy/layers/inet6.py

```diff
@@ -360,6 +360,11 @@ def default_payload_class(self,p):
             return Raw
         elif self.nh == 135 and len(p) > 3: # Mobile IPv6
             return _mip6_mhtype2cls.get(ord(p[2]), MIP6MH_Generic)
+        #FIXME
+        elif self.nh == 43: # Routing header
+            t = ord(p[2])
```

Comment:
Unless you plan to implement the other RO, it is better to remove the `routingtypescls`. If t == 4, you return IPv6ExtHdrSegmentRouting otherwise, you keep the current behavior.